main.go

```diff
@@ -1,48 +1,84 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"log"
 	"os"
 
 	"github.com/SUSE/scf-secret-generator/secrets"
 )
 
-func printHelp() {
-	fmt.Printf("Usage: %s <role-manifest>\n", os.Args[0])
-}
+var domain = flag.String(
+	"domain",
+	"",
+	"Domain",
+)
+
+var namespace = flag.String(
+	"namespace",
+	"",
+	"Kubernetes namespace",
+)
+
+var serviceDomainSuffix = flag.String(
+	"serviceDomainSuffix",
+	"",
+	"Service Domain Suffix",
+)
+
+var secretsName = flag.String(
+	"secretsName",
+	"",
+	"Secrets Name",
+)
+
+var secretsGeneration = flag.String(
+	"secretsGeneration",
+	"",
+	"Secrets Generation",
+)
+
+var certExpiration = flag.Int(
+	"certExpiration",
+	10950, // 30 years * 365 days
+	"Certificate expiration (in days)",
+)
 
 func main() {
-	if len(os.Args) != 2 {
-		printHelp()
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		fmt.Printf("Usage: %s <role-manifest>\n", os.Args[0])
+		flag.PrintDefaults()
```

Comment:
I think this means the first line is on `stdout`, but the rest is on `stderr`? It would probably make sense to output everything to the same place.